### PR TITLE
Add a new note about the variable product when the owner adds the fir…

### LINKED ISSUE #5934: New inbox message: Learn more about variable products
**Title:** Learn more about variable products
**Copy:** Variable products are a powerful product type that lets you offer a set of variations on a product, with control over prices, stock, image and more for each variation. They can be used for a product like a shirt, where you can offer a large, medium and small and in different colors.
**Segmentation:** new users
**Timing:** After adding the first product
**Cadence:** Once
**CTA leads to:** https://docs.woocommerce.com/document/variable-product/?utm_source=inbox
**CTA label:** Learn more

### PR DESCRIPTION
**Pull request by moon0326**
Fixes #5934 

This PR adds a note with a link to the variable product documentation when the store owner adds his/her first product.

### Screenshots

Screen Shot 2021-01-06 at 2 37 17 PM

### Detailed test instructions:

1. Add a new product if you don't have any yet.
2. Install & activate "WP Crontrol" plugin
3. Navigate to Tools -> Cron Events and run `wc_admin_daily` job
4. Navigate to WooCommerce -> Home and confirm a new note with the title "Learn more about variable products"

### CODE REVIEW

**Comment by becdetat** [comment]
This works well, however could this be triggered when the first product is actually added rather than waiting for the next `wc_admin_daily` cron run? That way the user would get much faster feedback.

**Comment by moon0326** [comment]
> This works well, however could this be triggered when the first product is actually added rather than waiting for the next `wc_admin_daily` cron run? That way the user would get much faster feedback.

That's a good point. I will see if there is a way to do it without affecting performance.

**Comment by moon0326** [comment]
@becdetat Changes are in https://github.com/woocommerce/woocommerce-admin/commit/b14b8ae5d98549c430e88b8bca9332e1c14d6aa2

**Review by louwie17 — CHANGES_REQUESTED** [review]
@moon0326 I was not able to get the Note to display with the latest changes, see comment (removing checking for $update fixed it for me). Maybe we should compare with the old status instead (if that's easily done).

A final question, should we still display the note if the user creates a Variation product as the first product? Or uploads product data that contains a variation product?
The former could make the note a bit confusing, this might be more likely with the new [product template feature](https://github.com/woocommerce/woocommerce-admin/pull/5892).
cc @pmcpinto for this question.

File: src/Notes/LearnMoreAboutVariableProducts.php
Comment (moon0326) [reply src/Notes/LearnMoreAboutVariableProducts.php:47]: Thank you @louwie17 for testing it. I will test it again and make sure it works as expected.

Comment (moon0326) [reply src/Notes/LearnMoreAboutVariableProducts.php]: @louwie17 Thank you for catching it. I was able to verify that `$update` is always true. I think I rushed my tests.

I made changes to check only `post_status` and `post_type` as that's enough for the purpose of this ticket. I also added a test case. Changes are in https://github.com/woocommerce/woocommerce-admin/commit/203569e0cc65782a0fe1c5aef2051c3050a5cc33

Comment (louwie17) [reply src/Notes/LearnMoreAboutVariableProducts.php]: Thanks @moon0326 that looks good!

File: src/Notes/LearnMoreAboutVariableProducts.php
Comment (louwie17) [reply src/Notes/LearnMoreAboutVariableProducts.php:1]: Nice work on the initial test, before merging, I would encourage to write two more test cases, for:
- Publishing a new post which isn't a product - should not show the note
- Saving a product as draft - should not show note
That way you got all the cases covered.

**Review by louwie17 — COMMENTED** [review]
@moon0326 Things tested well for me this time, thanks for the updates :)
I left [one comment](https://github.com/woocommerce/woocommerce-admin/pull/6021#discussion_r555871682) for a couple more test cases before merging, and to fix the merge conflict.

File: src/Notes/LearnMoreAboutVariableProducts.php
Comment (moon0326) [reply src/Notes/LearnMoreAboutVariableProducts.php:1]: @louwie17 thank you 👍  added two more test cases as suggested in https://github.com/woocommerce/woocommerce-admin/pull/6021/commits/87f2fda0fee5c730d11bfdf894474575469cedc3

Comment (louwie17) [reply src/Notes/LearnMoreAboutVariableProducts.php:1]: I appreciate the product template compliment 😊

Also thanks for writing the extra tests, those look good!

**Comment by pmcpinto** [comment]
> A final question, should we still display the note if the user creates a Variation product as the first product? Or uploads product data that contains a variation product?
> The former could make the note a bit confusing, this might be more likely with the new product template feature.
cc @pmcpinto for this question.

Good catch! Please add that exception to the message. Thanks!

**Comment by moon0326** [comment]
@louwie17 okay, updated the code to check for the product type. Changes are in https://github.com/woocommerce/woocommerce-admin/pull/6021/commits/fe8a38f811d7dec3910ff844e2aac7fce0359d61

File: src/Notes/LearnMoreAboutVariableProducts.php
Comment (louwie17) [inline src/Notes/LearnMoreAboutVariableProducts.php:36]: Nice find on this one

File: tests/notes/class-wc-tests-learn-more-about-variable-product.php
Comment (louwie17) [inline tests/notes/class-wc-tests-learn-more-about-variable-product.php:63]: Could you add a separate test for creating a `variable` product, and the note not showing?

Also maybe add this in-front of this test: `$wpdb->query( "delete from {$wpdb->prefix}wc_admin_notes" );`
So we know for sure that this test doesn't pass, because of the note already being present in the above test.

**Review by louwie17 — APPROVED** [review]
Nice work @moon0326! Code looks good, you got my approval on the basis of adding one more test case 😄 🧪 

Fun fact for variable products (don't worry about this use case):
- The product type does not become variable until you add at-least one variation, just selecting Variable in the product dropdown type keeps it as simple.

File: tests/notes/class-wc-tests-learn-more-about-variable-product.php
Comment (moon0326) [reply tests/notes/class-wc-tests-learn-more-about-variable-product.php:63]: @louwie17 thank you! I will add a variable test before merging 👍

**Comment by moon0326** [comment]
> Fun fact for variable products (don't worry about this use case):

Yup 😄 I also noticed it while testing. It looks like product variation creates a record in `wp_posts` and set its parent to the simple product. That was interesting to find out.

**Pull request merged**
